### PR TITLE
Update/jfalcou eve

### PR DESCRIPTION
**Pull request by jfalcou**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/jfalcou-eve/all/conanfile.py [hunk 1]
```diff
@@ -29,16 +33,24 @@ def _min_cppstd(self):
...
+            self.license = "MIT"
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.build.check_min_cppstd(self, self._min_cppstd)
-        if self.settings.compiler == "Visual Studio":
+        if self.info.settings.get_safe("compiler.cppstd"):
```
Comment (uilianries) [inline recipes/jfalcou-eve/all/conanfile.py:49]:
```suggestion
        # FIXME: Need to use self.info.settings for header-only
        if self.settings.get_safe("compiler.cppstd"):
```
So far, we can not use `self.info.settings` as compiler is empty and the CI is not able to detect the minimal compiler version.

File: recipes/jfalcou-eve/all/conanfile.py [hunk 2]
```diff
@@ -29,16 +33,24 @@ def _min_cppstd(self):
...
+            check_min_cppstd(self, self._min_cppstd)
+        if is_msvc(self):
             raise ConanInvalidConfiguration("EVE does not support MSVC yet (https://github.com/jfalcou/eve/issues/1022).")
-        if self.settings.compiler == "apple-clang":
+        if self.info.settings.compiler == "apple-clang":
```
Comment (uilianries) [inline recipes/jfalcou-eve/all/conanfile.py:53]:
```suggestion
        if self.settings.compiler == "apple-clang":
```

**Review by uilianries — COMMENTED** [review]
I've opened an issue to Conan client, to discuss our current situation, in case you want to follow. https://github.com/conan-io/conan/issues/12210

So far, we can't use `self.info.settings` with header-only, because the compiler will be empty when is not possible to check its minimal version. 

For Conan 2.x, it's expected `settings.compiler.cppstd` always configured, but on Conan 1.x is another reality.

**Review by uilianries — COMMENTED** [review]
With those suggestions, it should work

File: recipes/jfalcou-eve/all/test_v1_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package CXX)
+
+include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+conan_basic_setup(TARGETS)
+
+find_package(eve REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} ../../test_package.cpp)
```
Comment (uilianries) [inline recipes/jfalcou-eve/all/test_v1_package/CMakeLists.txt:9]:
```suggestion
add_executable(${PROJECT_NAME} ../test_package/test_package.cpp)
```

File: recipes/jfalcou-eve/all/conanfile.py [hunk 4]
```diff
@@ -1,25 +1,29 @@
...
     name = "jfalcou-eve"
     description = ("Expressive Velocity Engine - reimplementation of the old "
                    "Boost.SIMD on C++20"
                    )
-    homepage = "https://github.com/jfalcou/eve"
-    topics = ("c++", "simd")
-    license = "MIT"
+    homepage = "https://jfalcou.github.io/eve/"
+    topics = ("cpp", "simd")
+    license = ("BSL-1.0")
```
Comment (prince-chrismc) [inline recipes/jfalcou-eve/all/conanfile.py:22]:
```suggestion
    license = ("MIT", "BSL-1.0")
```

File: recipes/jfalcou-eve/all/conanfile.py [hunk 5]
```diff
@@ -29,14 +33,23 @@ def _min_cppstd(self):
...
+                "msvc": "1928",
                 "clang": "13",
-                "apple-clang": "13",
-                }
+                "apple-clang": "13"}
+
+    def configure(self):
+        version = Version(self.version.strip("v"))
+        if version.major < 2022 or (version.major == 2022 and version.minor < 9):
+            self.license = "MIT"
```
Comment (prince-chrismc) [inline recipes/jfalcou-eve/all/conanfile.py:43]:
```suggestion
        if version.major < 2022 or (version.major == 2022 and version.minor < 9):
            self.license = "MIT"
        else:
            self.license = "BSL-1.0"
```

**Review by prince-chrismc — COMMENTED** [review]
The recipe will expose the license for all versions while the package will have only one

**Review by prince-chrismc — DISMISSED** [review]

**Review by prince-chrismc — COMMENTED** [review]
whoops, see previous comments

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**